WordPress/gutenberg
# Remove obsolete wp-env configuration from package.json (#58877)

## What?
This PR removes obsolete `wp-env` configuration from the `package.json` file in the monorepo setup.

fixes #58877 

## Why?
The `wp-env` configuration in the `package.json` file is outdated and no longer necessary for the project. This PR aims to clean up the project by removing these obsolete settings, improving maintainability and reducing clutter in the configuration files.

## How?
The obsolete `wp-env` configuration settings, including 'plugin-dir', 'plugin-name', 'docker-template', 'welcome-logo', and 'welcome-build-command', are removed from the `package.json` file. Additionally, the referenced 'bin/docker-compose.override.yml.template' file is deleted as it is no longer in use.

## Testing Instructions
1. Clone the repository.
2. Checkout to the branch containing this PR.
3. Run `npm ci` to install dependencies.
4. Run `npm run build` to build the project.
5. Finally, run `npm run env start` to start WordPress including Gutenberg.
6. Verify that WordPress and Gutenberg work as expected without the removed `wp-env` configuration.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
N/A
